src/test/ui/unused-crate-deps/warn-attr.rs

```diff
@@ -0,0 +1,12 @@
+// Check for unused crate dep, no path
+
+// edition:2018
+// check-pass
+// aux-crate:bar=bar.rs
+// no-prefer-dynamic
```

Comment:
Unnecessary?
(This and other `no-prefer-dynamic`s.)

Reply:
If the `--crate-type lib` and `no-prefer-dynamic` were used to test the lint in different configurations, then it's better to merge all these tests into one.
We know that the lint is entirely unaffected by the crate types and linking types, and testing time is a finite resource.

Reply:
OK, agreed.

One other thing I couldn't set up in the test framework was transitive dependencies - I wanted to make a test verifying that if A depends on B and B on C, but C is unused, A gets no warnings. Can that be done? It didn't look like aux crates could have aux crates themselves.

Reply:
Looks like some UI tests work this way, e.g. `src\test\ui\static\auxiliary\static-priv-by-default.rs` depends on `src\test\ui\static\auxiliary\static_priv_by_default.rs`.